R/validate_per.R

```diff
@@ -0,0 +1,25 @@
+#' Validate per
+ 
+
+validate_per <- function(per = 'fortnight') {
+  if (missing(per)) {
```

Comment:
This is quite tricky but I'm intending `validate_per` to only be used in the body of functions. Since we will always write `validate_per(per)` the condition `missing(per)` *within* `validate_per` will never be true. We have to have an extra logical argument that checks for missingness *outside* the function, *i.e.*:

```r
validate_per <- function(per, per_missing) {
  if (per_missing) {
    message(...)
...
}
```

so that we can use it like 

```r
validate_per <- function(per = "fortnight", missing_per) {
  if (missing_per) {
    message("per missing...")
  }
  per <- per[1L]
  per
}

some_fn <- function(x, per = = c("fortnight", "year")) {
  per <- validate_per(per, missing(per))
}
```

Reply:
Do we need the `missing_per` argument if we always expect it to be there as its set by default in the function?

Reply:
Yes because the function won't know it's set at runtime.